### PR TITLE
Add mute source commands

### PR DESCRIPTION
**Pull request by mikhailswift**
Implements ToggleMute (inverts source's mute) and SetMute (sets mute to true or false).

### CODE REVIEW

File: WSRequestHandler.cpp [hunk 1]
```diff
@@ -378,17 +379,55 @@ void WSRequestHandler::HandleGetVolume(WSRequestHandler *owner) {
...
+void WSRequestHandler::ToggleMute(WSRequestHandler *owner) {
+	const char *item_name = obs_data_get_string(owner->_requestData, "source");
+	if (item_name == NULL) {
+		owner->SendErrorResponse("invalid request parameters");
+		return;
+	}
+
+	obs_source_t* item = obs_get_source_by_name(item_name);
+	if (!item) {
+		owner->SendErrorResponse("invalid request parameters");
```
Comment (Palakis) [inline WSRequestHandler.cpp]: Please add a `return` instruction after calling `SendErrorResponse`.

File: WSRequestHandler.cpp [hunk 2]
```diff
@@ -378,17 +379,55 @@ void WSRequestHandler::HandleGetVolume(WSRequestHandler *owner) {
...
+	const char *item_name = obs_data_get_string(owner->_requestData, "source");
+	bool mute = obs_data_get_bool(owner->_requestData, "mute");
+	if (item_name == NULL) {
+		owner->SendErrorResponse("invalid request parameters");
+		return;
+	}
+
+	obs_source_t* item = obs_get_source_by_name(item_name);
+	if (!item) {
+		owner->SendErrorResponse("specified source doesn't exist");
```
Comment (Palakis) [inline WSRequestHandler.cpp]: Same remark here as line 403.

File: PROTOCOL.md [hunk 3]
```diff
@@ -236,6 +236,20 @@ __Request fields__ :
 __Response__ : OK if source exists, with these additional fields :
 - **"name"** (string) : name of the requested source
 - **"volume"** (double) : volume of the requested source, on a linear scale (0.0 to 1.0)
+- **"muted"** (bool) : mute status of the requested source
+
+### "SetMute"
```
Comment (Palakis) [inline PROTOCOL.md]: The title is one level too high. Should be `#### "SetMute"` instead of `### "SetMute"`

File: PROTOCOL.md [hunk 4]
```diff
@@ -236,6 +236,20 @@ __Request fields__ :
...
+Mutes or unmutes a specific source.
+
+__Request fields__ :
+- **"source"** (string) : the name of the source
+- **"mute"** (bool) : the desired mute status
+
+### "ToggleMute"
```
Comment (Palakis) [inline PROTOCOL.md]: Same as line 241 : should be `#### "ToggleMute` instead.

**Comment by mikhailswift** [comment]
@Palakis Thanks for the review, and thanks for this great project!  This project has helped the stream group I'm involved with immensely.  I've addressed the items in your review.

**Comment by Huesdon** [comment]
Looking forward to this being packaged and released. I'm having to do mouseclicks to mute right now and it is painful. Thanks guys.

**Comment by Palakis** [comment]
Once this [nasty and tricky bug](https://github.com/Palakis/obs-websocket/issues/21) is fixed, I'll be able to package a new binary release.

**Pull request merged**